python/neuroglancer/pipeline/threaded_queue.py

```diff
@@ -1,211 +0,0 @@
-
-  def _consume_queue(self, terminate_evt):
-    """
-    This is the main thread function that consumes functions that are
-    inside the _queue object. To use, execute self._queue(fn), where fn
-    is a function that performs some kind of network IO or otherwise
-    benefits from threading and is independent.
-
-    terminate_evt is automatically passed in on thread creation and 
-    is a common event for this generation of threads. The threads
-    will terminate when the event is set and the queue burns down.
-
-    Returns: void
-    """
-    interface = self._initialize_interface()
-
-    while not terminate_evt.is_set():
-      try:
-        fn = self._queue.get(block=True, timeout=1)
-      except Queue.Empty:
-        continue # periodically check if the thread is supposed to die
-
-      fn = partial(fn, interface)
-
-      try:
-        self._consume_queue_execution(fn)
-      except Exception as err:
-        self._error_queue.put(err)
-
-  def _consume_queue_execution(self, fn):
-    """
-    The actual task execution in each thread. This
-    is broken out so that exceptions can be caught
-    in derived classes and allow them to manipulate 
-    the errant task, e.g. putting it back in the queue
-    for a retry.
-
-    Every task processed will automatically be marked complete.
-
-    Required:
-      [0] fn: A curried function that includes the interface
-              as its first argument.
-    Returns: void
-    """
-
-    # `finally` fires after all success or exceptions
-    # exceptions are handled in derived classes
-    # and uncaught ones are caught as a last resort
-    # in _consume_queue to be raised on the main thread.
-    try:
-      fn()
-    finally:
-      with self._processed_lock:
-        self.processed += 1
-        self._queue.task_done()
-
-  def wait(self):
-    """
-    Allow background threads to process until the
-    task queue is empty. If there are no threads,
-    in theory the queue should always be empty
-    as processing happens immediately on the main thread.
-
-    Required: None
-    
-    Returns: self (for chaining)
-
-    Raises: The first exception recieved from threads
-    """
-    if len(self._threads):
-        self._queue.join()
-
-    try:
-      # no blocking because we're guaranteed
-      # that all threads have finished processing
-      err = self._error_queue.get(block=False) 
-      self._error_queue.task_done()
-      raise err
-    except Queue.Empty:
-      pass
-
-    return self
-
-  def __del__(self):
-    self.wait() # if no threads were set the queue is always empty
-    self.kill_threads()
-
-  def __enter__(self):
-    if self.__class__ is ThreadedQueue and self._n_threads == 0:
-      raise ValueError("Using 0 threads in base class ThreadedQueue with statement will never exit.")
-
-    self.start_threads(self._n_threads)
-    return self
-
-  def __exit__(self, exception_type, exception_value, traceback):
-    self.wait()
-    self.kill_threads()
```

Comment:
Wasn't this file used by the task queue?

Reply:
Yea, I used the threaded_queue embedded in cloudvolume.